src/main/java/bio/terra/common/tracing/OkHttpTracingExtractor.java

```diff
@@ -0,0 +1,53 @@
+package bio.terra.common.tracing;
+
+import io.opencensus.contrib.http.HttpExtractor;
+import javax.annotation.Nullable;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class OkHttpTracingExtractor extends HttpExtractor<Request, Response> {
+
+  @Nullable
+  @Override
+  public String getRoute(Request request) {
+    // OpenCensus spec wants this to be something like the literal string
+    // "/api/workspace/{workspaceId}", whereas path would be "/api/workspace/12345".
+    // We don't have the route available and this is a best-effort method, so use path instead.
```

Comment:
Does this mean we'll get many more span categories than we would otherwise? How hard is it to aggregate across these?

What happens if you return `null` or an empty string for the route?

Reply:
These methods just populate the fields of a span which will later show up in GCP, e.g. this image. All fields are nullable here, so if you return `null` it just won't populate that field of the span. IMO either is fine, thinking about it more it's probably better to leave this blank than put a slightly-wrong value.